Mathlib/Tactic/Module.lean

```diff
@@ -0,0 +1,641 @@
+/-
+Copyright (c) 2024 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import Mathlib.Algebra.Algebra.Tower
+import Mathlib.GroupTheory.GroupAction.BigOperators
+import Mathlib.Tactic.Ring
+import Mathlib.Util.AtomM
+
+/-! # A tactic for normalization over modules
+
+This file provides the two tactics `match_scalars` and `module`.  Given a goal which is an equality
+in a type `M` (with `M` an `AddCommMonoid`), the `match_scalars` tactic parses the LHS and RHS of
+the goal as linear combinations of `M`-atoms over some semiring `R`, and reduces the goal to
+the respective equalities of the `R`-coefficients of each atom.  The `module` tactic does this and
+then runs the `ring` tactic on each of these coefficient-wise equalities, failing if this does not
+resolve them.
+
+The scalar type `R` is not pre-determined: instead it starts as `ℕ` (when each atom is initially
+given a scalar `(1:ℕ)`) and gets bumped up into bigger semirings when such semirings are
+encountered.  However, to permit this, it is assumed that there is a "linear order" on all the
+semirings which appear in the expression: for any two semirings `R` and `S` which occur, we have
+either `Algebra R S` or `Algebra S R`).
+-/
+
+open Lean hiding Module
+open Meta Elab Qq Mathlib.Tactic List
+
+namespace Mathlib.Tactic.Module
+
+/-! ### Theory of lists of pairs (scalar, vector)
+
+This section contains the lemmas which are orchestrated by the `match_scalars` and `module` tactics
+to prove goals in modules.  The basic object which these lemmas concern is `NF R M`, a type synonym
+for a list of ordered pairs in `R × M`, where typically `M` is an `R`-module.
+-/
+
+/-- Basic theoretical object of the `match_scalars` and `module` tactics: a type synonym for a list
```

Comment:
```suggestion
/-- Basic "Normal Form" object of the `match_scalars` and `module` tactics: a type synonym for a list
```
I think `NF` should be expanded to "Normal Form" somewhere in the first sentence.